### PR TITLE
add clojure challenge submission

### PR DESCRIPTION
**Pull request by mchampine**
Can't believe there wasn't a Clojure version already!

### CODE REVIEW

File: challenges/README.md [hunk 1]
```diff
@@ -54,11 +54,11 @@ var libraryData = {
                 "bookItems": [
                     {
                         "id": "book-item-1",
-                        "rackId": "rack-17",
+                        "rackId": "rack-17"
```
Comment (viebel) [inline challenges/README.md:57]: Thank you for the fixes

File: challenges/clojure/core.clj [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+(str/upper-case
+ (get-in ld ["catalog" "booksByIsbn" "978-1779501127" "title"]))
+;; => "WATCHMEN"
+
+;; Challenge #1 ‍
+;; Write a function named getBookProperty that receives library data
+;; and ISBN and a field name and returns the value of the field for
+;; the book with the given ISBN
+
+(defn getBookProperty
```
Comment (viebel) [inline challenges/clojure/core.clj:25]: In Clojure, we tend to prefer snake case for var names.
I know that in the instructions I used camelCase.
Would you like to transform var names to snake case?

Comment (mchampine) [reply challenges/clojure/core.clj:25]: I don't mind the break with Clojure convention really - at least for this purpose

**Comment by viebel** [comment]
@mchampine Please send me a message on Twitter to claim your free copy of the book.

**Pull request merged**